### DISCUSSION TITLE
Possibility to set "api/event" request timestamp

### DISCUSSION THREAD

**Discussion by clement-joye** [plausible/analytics#1973] [Anything Else]
Hello, I'm currently working on a way to defer the requests sent to Plausible as part of the concept of delayed computing.

Here's my use case: 
- I aggregate Plausible "events" in a queue every day (via Azure service bus or Azure Storage account for example)
- At the end of the day I run a "Time trigger" Azure function that will read/delete all events that have been queued for the day.
- Every event is then forwarded by the function to Plausible to process all events that occurred.

There are obviously pros and cons with this technique and I'm aware of that.

Now, the issue for the time being is that the requests to the "api/event" endpoint naturally don't support setting the timestamp ourselves, probably to avoid tampering with the data I imagine.

To workaround that, I modified lib\plausible_web\controllers\api\external_controller.ex in order to support setting the timestamp as shown below.
<img width="915" alt="image" src="https://user-images.githubusercontent.com/57441858/175810211-c5607ded-a7a1-4f7a-bda0-6973186f32a8.png">

My question is then, would it be possible to support setting the timestamp in the request? (maybe via environment variable and default being false.)

Thanks 
Clement

**Comment by ukutaht** [comment]
It's not as easy as you think.

Our session tracking currently only works real-time. We would have to build a completely new method for sessionization if we want to also ingest historical data.

In general I'm in favour of this but I don't see us finding time to work it anytime soon

**Reply by clement-joye** [reply]
Right, I was afraid that there would be more work needed than this. 
I suppose I don't really get all the features here with that, but at least the page views.
That's enough for my use case at the moment so i'll use that, but I'll keep an eye open at Plausible in the future anyhow :-) 

Thanks for taking the time @ukutaht !

**Reply by ukutaht** [reply]
Thanks @clement-joye !

**Reply by KaKi87** [reply]
Hi,

> In general I'm in favour of this but I don't see us finding time to work it anytime soon

Has there been any change regarding this ?

My use case is feeding data from from server-side access logs.

Thanks
